service/header/service.go

```diff
@@ -24,13 +26,17 @@ func NewHeaderService(
 	syncer *Syncer,
 	sub Subscriber,
 	p2pServer *P2PExchangeServer,
-	ex Exchange) *Service {
-	return &Service{
+	ex Exchange,
+	rpc *rpc.Server,
+) *Service {
+	serv := &Service{
 		syncer:    syncer,
 		sub:       sub,
 		p2pServer: p2pServer,
 		ex:        ex,
 	}
+	serv.RegisterEndpoints(rpc)
+	return serv
```

Comment:
Pls make registering on the Node's constructor. Like you do for ShareService and State